pkg/cloud/filter/filter_resource_test.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filter
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/compute/v1"
+)
+
+var (
+	projectID = os.Getenv("PROJECT_ID")
+	// test requires e2e cluster to be present in the project specified in order to match these filters
+	routeFilters   = map[string]interface{}{"name": "e2e-test-.*", "description": "k8s-node-route", "priority": 1000}
+	networkFilters = map[string]interface{}{"autoCreateSubnetworks": false}
+)
+
+func TestFilterResources(t *testing.T) {
+	t.Parallel()
+	if projectID == "" {
+		t.Skip("missing projectID... skipping test")
+	}
+
+	ctx := context.Background()
+
+	// See https://cloud.google.com/docs/authentication/.
+	// Use GOOGLE_APPLICATION_CREDENTIALS environment variable to specify
+	// a service account key file to authenticate to the API.
+	hc, err := google.DefaultClient(ctx, compute.ComputeScope)
+	if err != nil {
+		t.Errorf("Could not get authenticated client: %v", err)
+	}
+
+	svc, err := compute.New(hc)
+	if err != nil {
+		t.Errorf("Could not initialize gke client: %v", err)
```

Comment:
You use %v is some places and %s in other. Probably best to stay consistent to one. I would say %v since that is probably used more widely elsewhere.